client/src/routes/join/+page.svelte

```diff
@@ -44,11 +69,12 @@
       });
       // go to game page
       goto(`/game?code=${code}`);
-    } catch (err: any) {
+    } catch (err) {
       // if the lobby doesn't exist then error is thrown
+      const myError = err as Error;
       error = {
```

Comment:
Is this a safe type assertion? I feel like it might be better to use an if.

Reply:
the err parameter in the catch is either unknown or any so you literally have to tell typescript its type to be able to use it.

Reply:
the reason it is `unknown`  is because we actually don't know what it is. it might not be an error. in JavaScript you can throw any value.

here is what we can do:

at runtime check if it is an error using the JavaScript [instanceof operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/instanceof)

if it is an error then you can use it's message

if it is not an error then you still want to display something to the user so we can use the [String Constructor](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/String) without the `new` keyword which can convert any value to a string

we could use an if statement like this:

```ts
catch (err) {
  // if the lobby doesn't exist then error is thrown
  let errorMessage: string;
  if (err instanceof Error) {
    errorMessage = err.message;
  } else {
    errorMessage = String(err);
  }
  error = {
    status: true,
    message: errorMessage,
  };
  code = "";
}
```

or we can do it with the [ternary operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator) if you want
```ts
catch (err) {
  // if the lobby doesn't exist then error is thrown
  error = {
    status: true,
    message: err instanceof Error ? err.message : String(err),
  };
  code = "";
}
```

Reply:
Okay that makes sense for the unknown type